packages/core/src/routes/session.ts

```diff
@@ -34,11 +35,34 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
             passwordEncrypted && passwordEncryptionMethod && passwordEncryptionSalt,
             'session.invalid_sign_in_method'
           );
-          assertThat(
-            encryptPassword(id, password, passwordEncryptionSalt, passwordEncryptionMethod) ===
-              passwordEncrypted,
-            'session.invalid_credentials'
-          );
+
+          if (
```

Comment:
wondering if we should stick with the existing `assertThat` statement, with some work arounds. 
i.e. takes a callback method
or
as `assertThat` will throw an exception,  should we call the logging method in a catch block.
leave the final call to  @gao-sun

Reply:
we can `await` and `assertThat`, if you mean async operations.